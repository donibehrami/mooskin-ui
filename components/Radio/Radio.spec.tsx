import * as React from 'react';
import renderer from 'react-test-renderer';
import RadioGroup, {Radio} from './Radio';

import {shallow} from 'enzyme';

describe('Radio', () => {

    it('renders correctly', () => {
        const func = jest.fn();
        const func2 = jest.fn();

        const tree = renderer.create(
            <RadioGroup
                onChange={func}
                selected="radio1"
                dataLabel="plan"
                id="radio1"
                className="myClass"
                style={{color: 'blue'}}
                spacing={10}
                title={'Select a plan'}
                vertical
            >
                <Radio
                    value="radio1"
                    onClick={func2}
                    className="myClass"
                    style={{color: 'blue'}}
                    disabled
                    label="This is radio 1"
                    vertical
                    id={'radio1'}
                    name="radioName"
                />
                <Radio
                    value="radio2"
                    onClick={func2}
                    className="myClass"
                    style={{color: 'blue'}}
                    disabled
                    label="This is radio 2"
                    vertical
                    id={'radio2'}
                    name="radioName"
                />
            </RadioGroup>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders properly with 1 child', () => {

        const component = shallow(
            <RadioGroup selected="radio1" className="myClass" dataLabel="plan">
                <Radio value="radio1" id={'main'} label="dem labels" />
            </RadioGroup>
        );

        expect(component.find(Radio).length).toBe(1);
        expect(component.find(Radio).prop('value')).toEqual('radio1');
        expect(component.find(Radio).prop('label')).toEqual('dem labels');
        expect(component.find(Radio).prop('id')).toEqual('main');
        expect(component.find(Radio).prop('selected')).toEqual('radio1');
        expect(component.find('.myClass')).toBeTruthy;
        // expect(component.find('.myClass').prop('dataLabel')).toEqual('plan');
    });

    test('renders properly with multiple children', () => {

        const component = shallow(
            <RadioGroup selected="radio1">
                <Radio value="radio1" />
                <Radio value="radio2" />
                <Radio value="radio3" />
            </RadioGroup>
        );

        expect(component.find(Radio).length).toBe(3);
    });

    test('appends onClick callback to each Radio', () => {
        const func = jest.fn();

        const component = shallow(
            <RadioGroup onChange={func} selected="radio1" style={{color: 'blue'}}>
                <Radio value="radio1" />
                <Radio value="radio2" />
                <Radio value="radio3" />
            </RadioGroup>
        );

        // expect(component.find(RadioGroup).prop('style')).toEqual({color: 'blue'});
        component.find(Radio).forEach((radio) => {
            expect(radio.prop('onClick')).toBeTruthy();
        });
    });

    test('Option renders without onClick by default', () => {

        const component = shallow(<Radio value="radio1" />);

        expect(component.prop('onClick')).toBeFalsy();
    });

    test('callback func is called when Radio is clicked', () => {
        const func = jest.fn();

        const component = shallow(
            <RadioGroup onChange={func} >
                <Radio value="radio1" />
            </RadioGroup>
        );

        expect(component.find(Radio).simulate('click'));

        expect(func).toHaveBeenCalled();
    });

    test('callback func is not called when a disabled Radio is clicked', () => {
        const func = jest.fn();

        const component = shallow(
            <RadioGroup onChange={func}>
                <Radio value="radio1" disabled />
            </RadioGroup>
        );

        expect(component.find(Radio).prop('disabled')).toBeTruthy;
        expect(component.find(Radio).dive().find('input').simulate('click'));

        expect(func).not.toHaveBeenCalled();
    });
});
