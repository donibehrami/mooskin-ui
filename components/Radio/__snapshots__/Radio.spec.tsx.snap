// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders Radio correctly 1`] = `
<Radio
  className=""
  onClickRadio={[MockFunction]}
  selected={true}
  style={Object {}}
>
  <Component
    className=""
    onClickRadio={[MockFunction]}
    palette={
      Object {
        "backgroundColors": Object {
          "button": "#3fbaca",
          "common": "#ffffff",
          "disabledButton": "#e2e2e2",
          "main": "#f7f7f7",
          "secondaryToggle": "#ffffff",
          "toggle": "#425579",
        },
        "borderColors": Object {
          "common": "#e2e2e2",
        },
        "fontColors": Object {
          "anchor": "#3fbaca",
          "button": "#ffffff",
          "common": "#293346",
          "description": "#9d9d9d",
          "disabledAnchor": "#9d9d9d",
          "label": "#2d2d2d",
          "secondaryToggle": "#9d9d9d",
          "text": "#2d2d2d",
          "toggle": "#ffffff",
        },
      }
    }
    selected={true}
    style={Object {}}
  >
    <StyledRadio
      className=""
      onClickRadio={[MockFunction]}
      palette={
        Object {
          "backgroundColors": Object {
            "button": "#3fbaca",
            "common": "#ffffff",
            "disabledButton": "#e2e2e2",
            "main": "#f7f7f7",
            "secondaryToggle": "#ffffff",
            "toggle": "#425579",
          },
          "borderColors": Object {
            "common": "#e2e2e2",
          },
          "fontColors": Object {
            "anchor": "#3fbaca",
            "button": "#ffffff",
            "common": "#293346",
            "description": "#9d9d9d",
            "disabledAnchor": "#9d9d9d",
            "label": "#2d2d2d",
            "secondaryToggle": "#9d9d9d",
            "text": "#2d2d2d",
            "toggle": "#ffffff",
          },
        }
      }
      selected={true}
      style={Object {}}
    >
      <Box
        className="sc-hKFyIo dxNlPq"
        onClickRadio={[MockFunction]}
        palette={
          Object {
            "backgroundColors": Object {
              "button": "#3fbaca",
              "common": "#ffffff",
              "disabledButton": "#e2e2e2",
              "main": "#f7f7f7",
              "secondaryToggle": "#ffffff",
              "toggle": "#425579",
            },
            "borderColors": Object {
              "common": "#e2e2e2",
            },
            "fontColors": Object {
              "anchor": "#3fbaca",
              "button": "#ffffff",
              "common": "#293346",
              "description": "#9d9d9d",
              "disabledAnchor": "#9d9d9d",
              "label": "#2d2d2d",
              "secondaryToggle": "#9d9d9d",
              "text": "#2d2d2d",
              "toggle": "#ffffff",
            },
          }
        }
        selected={true}
        style={Object {}}
      >
        <Component
          className="sc-hKFyIo dxNlPq"
          onClickRadio={[MockFunction]}
          palette={
            Object {
              "backgroundColors": Object {
                "button": "#3fbaca",
                "common": "#ffffff",
                "disabledButton": "#e2e2e2",
                "main": "#f7f7f7",
                "secondaryToggle": "#ffffff",
                "toggle": "#425579",
              },
              "borderColors": Object {
                "common": "#e2e2e2",
              },
              "fontColors": Object {
                "anchor": "#3fbaca",
                "button": "#ffffff",
                "common": "#293346",
                "description": "#9d9d9d",
                "disabledAnchor": "#9d9d9d",
                "label": "#2d2d2d",
                "secondaryToggle": "#9d9d9d",
                "text": "#2d2d2d",
                "toggle": "#ffffff",
              },
            }
          }
          selected={true}
          style={Object {}}
        >
          <StyledBox
            className="sc-hKFyIo dxNlPq"
            onClickRadio={[MockFunction]}
            palette={
              Object {
                "backgroundColors": Object {
                  "button": "#3fbaca",
                  "common": "#ffffff",
                  "disabledButton": "#e2e2e2",
                  "main": "#f7f7f7",
                  "secondaryToggle": "#ffffff",
                  "toggle": "#425579",
                },
                "borderColors": Object {
                  "common": "#e2e2e2",
                },
                "fontColors": Object {
                  "anchor": "#3fbaca",
                  "button": "#ffffff",
                  "common": "#293346",
                  "description": "#9d9d9d",
                  "disabledAnchor": "#9d9d9d",
                  "label": "#2d2d2d",
                  "secondaryToggle": "#9d9d9d",
                  "text": "#2d2d2d",
                  "toggle": "#ffffff",
                },
              }
            }
            selected={true}
            style={Object {}}
          >
            <div
              className="sc-bdnylx dyDpKS sc-hKFyIo dxNlPq"
              onClickRadio={[MockFunction]}
              selected={true}
              style={Object {}}
            >
              <RadioIcon
                className=""
                color="red"
                key="0/.0"
                onClick={[Function]}
                style={Object {}}
              >
                <Component
                  className=""
                  color="red"
                  onClick={[Function]}
                  palette={
                    Object {
                      "backgroundColors": Object {
                        "button": "#3fbaca",
                        "common": "#ffffff",
                        "disabledButton": "#e2e2e2",
                        "main": "#f7f7f7",
                        "secondaryToggle": "#ffffff",
                        "toggle": "#425579",
                      },
                      "borderColors": Object {
                        "common": "#e2e2e2",
                      },
                      "fontColors": Object {
                        "anchor": "#3fbaca",
                        "button": "#ffffff",
                        "common": "#293346",
                        "description": "#9d9d9d",
                        "disabledAnchor": "#9d9d9d",
                        "label": "#2d2d2d",
                        "secondaryToggle": "#9d9d9d",
                        "text": "#2d2d2d",
                        "toggle": "#ffffff",
                      },
                    }
                  }
                  style={Object {}}
                >
                  <StyledRadioIcon
                    className=""
                    color="red"
                    onClick={[Function]}
                    palette={
                      Object {
                        "backgroundColors": Object {
                          "button": "#3fbaca",
                          "common": "#ffffff",
                          "disabledButton": "#e2e2e2",
                          "main": "#f7f7f7",
                          "secondaryToggle": "#ffffff",
                          "toggle": "#425579",
                        },
                        "borderColors": Object {
                          "common": "#e2e2e2",
                        },
                        "fontColors": Object {
                          "anchor": "#3fbaca",
                          "button": "#ffffff",
                          "common": "#293346",
                          "description": "#9d9d9d",
                          "disabledAnchor": "#9d9d9d",
                          "label": "#2d2d2d",
                          "secondaryToggle": "#9d9d9d",
                          "text": "#2d2d2d",
                          "toggle": "#ffffff",
                        },
                      }
                    }
                    style={Object {}}
                  >
                    <Box
                      className="sc-eCApGN eSmmaN"
                      color="red"
                      onClick={[Function]}
                      palette={
                        Object {
                          "backgroundColors": Object {
                            "button": "#3fbaca",
                            "common": "#ffffff",
                            "disabledButton": "#e2e2e2",
                            "main": "#f7f7f7",
                            "secondaryToggle": "#ffffff",
                            "toggle": "#425579",
                          },
                          "borderColors": Object {
                            "common": "#e2e2e2",
                          },
                          "fontColors": Object {
                            "anchor": "#3fbaca",
                            "button": "#ffffff",
                            "common": "#293346",
                            "description": "#9d9d9d",
                            "disabledAnchor": "#9d9d9d",
                            "label": "#2d2d2d",
                            "secondaryToggle": "#9d9d9d",
                            "text": "#2d2d2d",
                            "toggle": "#ffffff",
                          },
                        }
                      }
                      style={Object {}}
                    >
                      <Component
                        className="sc-eCApGN eSmmaN"
                        color="red"
                        onClick={[Function]}
                        palette={
                          Object {
                            "backgroundColors": Object {
                              "button": "#3fbaca",
                              "common": "#ffffff",
                              "disabledButton": "#e2e2e2",
                              "main": "#f7f7f7",
                              "secondaryToggle": "#ffffff",
                              "toggle": "#425579",
                            },
                            "borderColors": Object {
                              "common": "#e2e2e2",
                            },
                            "fontColors": Object {
                              "anchor": "#3fbaca",
                              "button": "#ffffff",
                              "common": "#293346",
                              "description": "#9d9d9d",
                              "disabledAnchor": "#9d9d9d",
                              "label": "#2d2d2d",
                              "secondaryToggle": "#9d9d9d",
                              "text": "#2d2d2d",
                              "toggle": "#ffffff",
                            },
                          }
                        }
                        style={Object {}}
                      >
                        <StyledBox
                          className="sc-eCApGN eSmmaN"
                          color="red"
                          onClick={[Function]}
                          palette={
                            Object {
                              "backgroundColors": Object {
                                "button": "#3fbaca",
                                "common": "#ffffff",
                                "disabledButton": "#e2e2e2",
                                "main": "#f7f7f7",
                                "secondaryToggle": "#ffffff",
                                "toggle": "#425579",
                              },
                              "borderColors": Object {
                                "common": "#e2e2e2",
                              },
                              "fontColors": Object {
                                "anchor": "#3fbaca",
                                "button": "#ffffff",
                                "common": "#293346",
                                "description": "#9d9d9d",
                                "disabledAnchor": "#9d9d9d",
                                "label": "#2d2d2d",
                                "secondaryToggle": "#9d9d9d",
                                "text": "#2d2d2d",
                                "toggle": "#ffffff",
                              },
                            }
                          }
                          style={Object {}}
                        >
                          <div
                            className="sc-bdnylx dNqqpZ sc-eCApGN eSmmaN"
                            color="red"
                            onClick={[Function]}
                            style={Object {}}
                          >
                            radio_button_checked
                          </div>
                        </StyledBox>
                      </Component>
                    </Box>
                  </StyledRadioIcon>
                </Component>
              </RadioIcon>
              <RadioLabel
                className=""
                key="1/.1"
                onClick={[Function]}
                style={Object {}}
              >
                <Component
                  className=""
                  onClick={[Function]}
                  palette={
                    Object {
                      "backgroundColors": Object {
                        "button": "#3fbaca",
                        "common": "#ffffff",
                        "disabledButton": "#e2e2e2",
                        "main": "#f7f7f7",
                        "secondaryToggle": "#ffffff",
                        "toggle": "#425579",
                      },
                      "borderColors": Object {
                        "common": "#e2e2e2",
                      },
                      "fontColors": Object {
                        "anchor": "#3fbaca",
                        "button": "#ffffff",
                        "common": "#293346",
                        "description": "#9d9d9d",
                        "disabledAnchor": "#9d9d9d",
                        "label": "#2d2d2d",
                        "secondaryToggle": "#9d9d9d",
                        "text": "#2d2d2d",
                        "toggle": "#ffffff",
                      },
                    }
                  }
                  style={Object {}}
                >
                  <StyledRadioLabel
                    className=""
                    onClick={[Function]}
                    palette={
                      Object {
                        "backgroundColors": Object {
                          "button": "#3fbaca",
                          "common": "#ffffff",
                          "disabledButton": "#e2e2e2",
                          "main": "#f7f7f7",
                          "secondaryToggle": "#ffffff",
                          "toggle": "#425579",
                        },
                        "borderColors": Object {
                          "common": "#e2e2e2",
                        },
                        "fontColors": Object {
                          "anchor": "#3fbaca",
                          "button": "#ffffff",
                          "common": "#293346",
                          "description": "#9d9d9d",
                          "disabledAnchor": "#9d9d9d",
                          "label": "#2d2d2d",
                          "secondaryToggle": "#9d9d9d",
                          "text": "#2d2d2d",
                          "toggle": "#ffffff",
                        },
                      }
                    }
                    style={Object {}}
                  >
                    <Label
                      className="sc-jSFkmK kyZcBf"
                      onClick={[Function]}
                      palette={
                        Object {
                          "backgroundColors": Object {
                            "button": "#3fbaca",
                            "common": "#ffffff",
                            "disabledButton": "#e2e2e2",
                            "main": "#f7f7f7",
                            "secondaryToggle": "#ffffff",
                            "toggle": "#425579",
                          },
                          "borderColors": Object {
                            "common": "#e2e2e2",
                          },
                          "fontColors": Object {
                            "anchor": "#3fbaca",
                            "button": "#ffffff",
                            "common": "#293346",
                            "description": "#9d9d9d",
                            "disabledAnchor": "#9d9d9d",
                            "label": "#2d2d2d",
                            "secondaryToggle": "#9d9d9d",
                            "text": "#2d2d2d",
                            "toggle": "#ffffff",
                          },
                        }
                      }
                      style={Object {}}
                    >
                      <Component
                        className="sc-jSFkmK kyZcBf"
                        onClick={[Function]}
                        palette={
                          Object {
                            "backgroundColors": Object {
                              "button": "#3fbaca",
                              "common": "#ffffff",
                              "disabledButton": "#e2e2e2",
                              "main": "#f7f7f7",
                              "secondaryToggle": "#ffffff",
                              "toggle": "#425579",
                            },
                            "borderColors": Object {
                              "common": "#e2e2e2",
                            },
                            "fontColors": Object {
                              "anchor": "#3fbaca",
                              "button": "#ffffff",
                              "common": "#293346",
                              "description": "#9d9d9d",
                              "disabledAnchor": "#9d9d9d",
                              "label": "#2d2d2d",
                              "secondaryToggle": "#9d9d9d",
                              "text": "#2d2d2d",
                              "toggle": "#ffffff",
                            },
                          }
                        }
                        style={Object {}}
                      >
                        <StyledLabel
                          boxAs="label"
                          className="sc-jSFkmK kyZcBf"
                          onClick={[Function]}
                          palette={
                            Object {
                              "backgroundColors": Object {
                                "button": "#3fbaca",
                                "common": "#ffffff",
                                "disabledButton": "#e2e2e2",
                                "main": "#f7f7f7",
                                "secondaryToggle": "#ffffff",
                                "toggle": "#425579",
                              },
                              "borderColors": Object {
                                "common": "#e2e2e2",
                              },
                              "fontColors": Object {
                                "anchor": "#3fbaca",
                                "button": "#ffffff",
                                "common": "#293346",
                                "description": "#9d9d9d",
                                "disabledAnchor": "#9d9d9d",
                                "label": "#2d2d2d",
                                "secondaryToggle": "#9d9d9d",
                                "text": "#2d2d2d",
                                "toggle": "#ffffff",
                              },
                            }
                          }
                          style={Object {}}
                        >
                          <Box
                            boxAs="label"
                            className="sc-dlnjPT iSwLvp sc-jSFkmK kyZcBf"
                            onClick={[Function]}
                            palette={
                              Object {
                                "backgroundColors": Object {
                                  "button": "#3fbaca",
                                  "common": "#ffffff",
                                  "disabledButton": "#e2e2e2",
                                  "main": "#f7f7f7",
                                  "secondaryToggle": "#ffffff",
                                  "toggle": "#425579",
                                },
                                "borderColors": Object {
                                  "common": "#e2e2e2",
                                },
                                "fontColors": Object {
                                  "anchor": "#3fbaca",
                                  "button": "#ffffff",
                                  "common": "#293346",
                                  "description": "#9d9d9d",
                                  "disabledAnchor": "#9d9d9d",
                                  "label": "#2d2d2d",
                                  "secondaryToggle": "#9d9d9d",
                                  "text": "#2d2d2d",
                                  "toggle": "#ffffff",
                                },
                              }
                            }
                            style={Object {}}
                          >
                            <Component
                              boxAs="label"
                              className="sc-dlnjPT iSwLvp sc-jSFkmK kyZcBf"
                              onClick={[Function]}
                              palette={
                                Object {
                                  "backgroundColors": Object {
                                    "button": "#3fbaca",
                                    "common": "#ffffff",
                                    "disabledButton": "#e2e2e2",
                                    "main": "#f7f7f7",
                                    "secondaryToggle": "#ffffff",
                                    "toggle": "#425579",
                                  },
                                  "borderColors": Object {
                                    "common": "#e2e2e2",
                                  },
                                  "fontColors": Object {
                                    "anchor": "#3fbaca",
                                    "button": "#ffffff",
                                    "common": "#293346",
                                    "description": "#9d9d9d",
                                    "disabledAnchor": "#9d9d9d",
                                    "label": "#2d2d2d",
                                    "secondaryToggle": "#9d9d9d",
                                    "text": "#2d2d2d",
                                    "toggle": "#ffffff",
                                  },
                                }
                              }
                              style={Object {}}
                            >
                              <StyledBox
                                as="label"
                                boxAs="label"
                                className="sc-dlnjPT iSwLvp sc-jSFkmK kyZcBf"
                                onClick={[Function]}
                                palette={
                                  Object {
                                    "backgroundColors": Object {
                                      "button": "#3fbaca",
                                      "common": "#ffffff",
                                      "disabledButton": "#e2e2e2",
                                      "main": "#f7f7f7",
                                      "secondaryToggle": "#ffffff",
                                      "toggle": "#425579",
                                    },
                                    "borderColors": Object {
                                      "common": "#e2e2e2",
                                    },
                                    "fontColors": Object {
                                      "anchor": "#3fbaca",
                                      "button": "#ffffff",
                                      "common": "#293346",
                                      "description": "#9d9d9d",
                                      "disabledAnchor": "#9d9d9d",
                                      "label": "#2d2d2d",
                                      "secondaryToggle": "#9d9d9d",
                                      "text": "#2d2d2d",
                                      "toggle": "#ffffff",
                                    },
                                  }
                                }
                                style={Object {}}
                              >
                                <label
                                  className="sc-bdnylx dyDpKS sc-dlnjPT iSwLvp sc-jSFkmK kyZcBf"
                                  onClick={[Function]}
                                  style={Object {}}
                                >
                                  Normal Radio
                                </label>
                              </StyledBox>
                            </Component>
                          </Box>
                        </StyledLabel>
                      </Component>
                    </Label>
                  </StyledRadioLabel>
                </Component>
              </RadioLabel>
            </div>
          </StyledBox>
        </Component>
      </Box>
    </StyledRadio>
  </Component>
</Radio>
`;
